tests/test_distortion_map_creation.py

```diff
@@ -0,0 +1,119 @@
+    rng = np.random.default_rng(seed=17)
+    select_stars = rng.choice(len(matches), size=150, replace=False)
+
+    astrom_cal = astrom.boresight_calibration(input_dataset=dataset, field_path=field_path, field_matches=[matches[select_stars]], find_distortion=True)
+
+    ## check that the distortion map does not create offsets greater than 4[mas]
+        # compute the distortion maps created from the best fit coeffs
+    coeffs = astrom_cal.distortion_coeffs
+    expected_coeffs = np.genfromtxt(distortion_coeffs_path)
+
+        # note the image shape and center around the image center
+    image_shape = np.shape(dataset[0].data)
+    yorig, xorig = np.indices(image_shape)
+    y0, x0 = image_shape[0]//2, image_shape[1]//2
+    yorig -= y0
+    xorig -= x0
+
+        # get the number of fitting params from the order
+    fitorder = int(coeffs[1])
+    fitparams = (fitorder + 1)**2
+    true_fitorder = int(expected_coeffs[-1])
+    true_fitparams = (true_fitorder + 1)**2
+
+        # reshape the coeff arrays
+    best_params_x = coeffs[0][:fitparams]
+    best_params_x = best_params_x.reshape(fitorder+1, fitorder+1)
+    total_orders = np.arange(fitorder+1)[:,None] + np.arange(fitorder+1)[None, :]
+    best_params_x = best_params_x / 500**(total_orders)
+
+    true_params_x = expected_coeffs[:-1][:true_fitparams]
+    true_params_x = true_params_x.reshape(true_fitorder+1, true_fitorder+1)
+    true_total_orders = np.arange(true_fitorder+1)[:,None] + np.arange(true_fitorder+1)[None, :]
+    true_params_x = true_params_x / 500**(true_total_orders)
+
+        # evaluate the polynomial at all pixel positions
+    x_corr = np.polynomial.legendre.legval2d(xorig.ravel(), yorig.ravel(), best_params_x)
+    x_corr = x_corr.reshape(xorig.shape)
+    x_diff = x_corr - xorig
+
+    true_x_corr = np.polynomial.legendre.legval2d(xorig.ravel(), yorig.ravel(), true_params_x)
+    true_x_corr = true_x_corr.reshape(xorig.shape)
+    true_x_diff = true_x_corr - xorig
+
+        # reshape and evaluate the same for y
+    best_params_y = coeffs[0][fitparams:]
+    best_params_y = best_params_y.reshape(fitorder+1, fitorder+1)
+    best_params_y = best_params_y / 500**(total_orders)
+
+    true_params_y = expected_coeffs[:-1][true_fitparams:]
+    true_params_y = true_params_y.reshape(true_fitorder+1, true_fitorder+1)
+    true_params_y = true_params_y / 500**(true_total_orders)
+    
+        # evaluate the polynomial at all pixel positions
+    y_corr = np.polynomial.legendre.legval2d(xorig.ravel(), yorig.ravel(), best_params_y)
+    y_corr = y_corr.reshape(yorig.shape)
+    y_diff = y_corr - yorig
+
+    true_y_corr = np.polynomial.legendre.legval2d(xorig.ravel(), yorig.ravel(), true_params_y)
+    true_y_corr = true_y_corr.reshape(yorig.shape)
+    true_y_diff = true_y_corr - yorig
+
+    # check the distortion maps are less than the maximum injected distortion (~3 pixels)
+    assert np.all(np.abs(x_diff)) < np.max(np.abs(true_x_diff))
```

Comment:
Perhaps we should follow this statement from the Zellem paper: "Current model distortion maps indicate that distortion is less than 4 mas in the central 1 arcsec × 1 arcsec (TTR5) region of the detector". So check that the error in the distortion within the central 1" x 1" region is < 4 mas.